testsuite/tests/lib-filename/temp.ml

```diff
@@ -0,0 +1,23 @@
+(* TEST
+*)
+
+let () =
+  let tmpdir1 = ref "" in
+  let tmpdir2 = ref "" in
+  let passed = try
+    tmpdir1 := Filename.temp_dir "morx" "fleem";
+    assert (Sys.is_directory !tmpdir1);
+    tmpdir2 := Filename.temp_dir "morx" "fleem" ;
+    assert (Sys.is_directory !tmpdir2);
+    assert (!tmpdir1 <> !tmpdir2);
+    true
+  with _ ->
+    if !tmpdir1 <> "" then
+      Sys.rmdir !tmpdir1
+   else ();
+    if !tmpdir2 <> "" then
+      Sys.rmdir !tmpdir2
+   else ();
+  false
+  in
+  assert passed
```

Comment:
This test seems reasonable, but has a number of issues:
- strange indentation
- `else ()` is fairly unusual
- cleans up tmp dirs on error but not on success
- failures in this test would contain no information, because it swalows errors

I'd recommend writing this as:
```ocaml
 let tmpdir1 = ref "" in
 let tmpdir2 = ref "" in
 Fun.protect ~finally:(fun () ->
   if !tmpdir1 <> "" then Sys.rmdir !tmpdir1;
   if !tmpdir2 <> "" then Sys.rmdir !tmpdir2;
 ) (fun () ->
    tmpdir1 := Filename.temp_dir "morx" "fleem";
    assert (Sys.is_directory !tmpdir1);
    tmpdir2 := Filename.temp_dir "morx" "fleem" ;
    assert (Sys.is_directory !tmpdir2);
    assert (!tmpdir1 <> !tmpdir2);
  )
```